pareto/utilities/cm_utils/run_utils.py

```diff
@@ -0,0 +1,432 @@
+        if pyo.check_optimal_termination(model.status):
+            print_results_summary(model)
+        else:
+            print(f"... no feasible solution found in {max_iterations} iterations")
+    # #displays the data from each model
+    # for node_num in final_values:
+    #     print(f"\n----------------model using: {node_num}----------------\n")
+    #     if final_values[node_num] == None:
+    #         print("model is infeasible.")
+    #         continue
+    #     for value in final_values[node_num]:
+    #         print(f"{value}: {final_values[node_num][value]}")
+    #     #plots the final values using matplotlib
+    #     print("------------------------------------------------")
+
+    # grabs total cost of every node
+    total_costs = [
+        (value, final_values[value]["\nTotal cost"]) for value in final_values
+    ]
+    sorted_costs = sorted(total_costs, key=lambda x: x[1])
+
+    sorted_costs_nozero = [
+        (value, total_cost) for (value, total_cost) in sorted_costs if total_cost > 1e-4
+    ]
+    min_node, min_cost = sorted_costs_nozero[0]
+
+    # plot comparisons between nodes
+    plot_comparisons(final_values, sorted_costs)
+
+    # plot schematic
+    # plot_schematic(df_parameters, total_costs)
+
+    # print node with smallest total cost
+    print(
+        f"\nNode {min_node} had the smallest total cost, which amounted to {min_cost}."
+    )
+
+    return min_node, models
+
+
+def solving(
+    model,
+    max_iterations=3000,
+    tee=False,
+    inf_recs=False,
+):
+    """
+    This function solves the model in the following manner:
+    Flow LP -> bilinear NLP
+    After solving a summary report of the cost breakdown
+    and the model itself is returned.
+
+    Conc LP and 0 flows fixed NLP solves have been commented out.
+    If required these sections can be uncommented and used to initialize
+    to the final bilinear NLP
+
+    Arguments:
+    model: The model to run
+    max_iterations: The maximum number of iterations ipopt will run for
+
+    Returns:
+    model: Solved model
+    values: Dictionary of broken down costs
+    """
+    # # ------------------------------Flow based LP------------------------------------
+
+    # fixing concentration and making the model linear
+    model = alter_nonlinear_cons(model, deactivate=True)
+
+    # running linear flow model
+    print("\nDeveloping an initialization...")
+    opt = pyo.SolverFactory("ipopt")
+    # opt.options['NonConvex'] = 2
+    # opt.options['TimeLimit'] = 150
+    status = opt.solve(model, tee=False)
+    term_cond = pyo.check_optimal_termination(status)
+    if not term_cond:
+        print(
+            "Linear model is infeasibility. This likely means the production flow \
+              exceeds the disposal and treatment capacities"
+        )
+
+    # unfixing all the initialized variables
+    # model = obj_fix(model, [], deactivate=[], activate= conclist+flowlist)
```

Comment:
```suggestion
```
Please remove or uncomment as appropriate.

Reply:
With Travis' comments addressed, the only other thing I would add is that the folder structure is integrated in an unintuitive way. The CM_module is stored in other_models, but its utilities are stored in the utilities folder, for example.

Rather than the partial integration into the PARETO file structure, why not just keep everything contained in the CM_module, with its own internal file structure, i.e.,

CM_module
-> notebooks
-> utilities
-> scripts (etc.)